browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -255,10 +256,24 @@ void BraveShieldsWebContentsObserver::ReadyToCommitNavigation(
       });
 }
 
+void BraveShieldsWebContentsObserver::BlockAllowedScripts(
+    const std::vector<std::string>& scripts) {
+  for (const auto& script : scripts) {
+    auto origin = url::Origin::Create(GURL(script));
+    bool is_origin = origin.Serialize() == script;
+    base::EraseIf(allowed_scripts_, [is_origin, script,
+                                     origin](const std::string& value) {
+      // scripts array may have both origins or full scripts paths.
```

Comment:
`SetAllowScriptsFromOriginsOnce` only matches based on origin so having a full script url in this list seems confusing at best